src/components/ScrollHandler/ScrollHandler.js

```diff
@@ -0,0 +1,26 @@
+import { useEffect } from 'react';
+import PropTypes from 'prop-types';
+import { withRouter } from 'react-router-dom';
+
+const ScrollHandler = ({ location }) => {
```

Comment:
Fyi: There is also the package `react-router-hash-link`. Any reason in particular why you didn't use that? Seems to be the first answer on stackoverflow